lib/src/mount.rs

```diff
@@ -50,13 +58,42 @@ fn run_findmnt(args: &[&str], path: &str) -> Result<Filesystem> {
 /// Inspect a target which must be a mountpoint root - it is an error
 /// if the target is not the mount root.
 pub(crate) fn inspect_filesystem(path: &Utf8Path) -> Result<Filesystem> {
-    run_findmnt(&["--mountpoint"], path.as_str())
+    findmnt_filesystem(&["--mountpoint"], path.as_str())
 }
 
 #[context("Inspecting filesystem by UUID {uuid}")]
 /// Inspect a filesystem by partition UUID
 pub(crate) fn inspect_filesystem_by_uuid(uuid: &str) -> Result<Filesystem> {
-    run_findmnt(&["--source"], &(format!("UUID={uuid}")))
+    findmnt_filesystem(&["--source"], &(format!("UUID={uuid}")))
+}
+
+// Check if a specified device contains an already mounted filesystem
+pub(crate) fn is_mounted_in_pid1_mountns(path: String) -> Result<bool> {
+    let o = run_findmnt(&["-N"], "1")?;
+    for fs in o.filesystems.iter() {
+        if is_source_mounted(&path, fs)? {
+            return Ok(true);
+        }
+    }
+
+    Ok(false)
+}
+
+// Recursively check a given filesystem to see if it contains an already mounted source
+pub(crate) fn is_source_mounted(path: &String, mounted_fs: &Filesystem) -> Result<bool> {
```

Comment:
- This function looks infallible so I think it can just be `-> bool`
- Prefer using `&str` vs `&String` for parameters; recent blog: https://steveklabnik.com/writing/when-should-i-use-string-vs-str/